### PR TITLE
[Web] Toolbar UI language duplication fix

### LINKED ISSUE #1275: [Web] Language picker in toolbar shows repeated languages
image

It appears that these languages have different region or script codes on them. We need to either differentiate or collapse them (probably collapse)

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #1275.

Now that KeymanWeb is using BCP-47 codes to represent languages, it's quite possible for more than one language code to map to a single keyboard.  This was the cause of the language duplication problem on the Toolbar UI's regions.  This fix punts that to the keyboard selection drop-down, which in turns allows the user to select which language-code variant of the keyboard to use, providing differentiation between/among the duplicates.

### CODE REVIEW

File: web/source/kmwutils.ts [hunk 1]
```diff
@@ -51,6 +51,14 @@ namespace com.keyman {
...
     }
 
+    ['getBCP47Components'](lgCode: string): string[] {
+      if(lgCode.indexOf('-')==-1) {
+        return [lgCode];
+      } else {
+        return lgCode.split('-');
+      }
+    }
+
```
Comment (darcywong00) [inline web/source/kmwutils.ts]: This is fine for getting the language code portion of the BCP 47 tag. (Rename to `getLanguageCode`?)

Later on if we need to parse out the script, region, variant components, maybe we should see what's available on npm.
(Off a quick search, maybe [language-tags](https://www.npmjs.com/package/language-tags)?)

File: web/testing/toolbar/index.html [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    <!-- Set the viewport width to match phone and tablet device widths -->                         
+    <meta name="viewport" content="width=device-width,user-scalable=no" /> 
+
+    <!-- Allow KeymanWeb to be saved to the iPhone home screen -->
+    <meta name="apple-mobile-web-app-capable" content="yes" />
+    
+    <!-- Enable IE9 Standards mode -->
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" /> 
+      
+    <title>KeymanWeb Sample Page - Unminified Source</title>
```
Comment (darcywong00) [inline web/testing/toolbar/index.html]: `<title>Toolbar UI Testing</title>`

File: web/testing/toolbar/index.html [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+    <div>
+    <!-- 
+      The following DIV is used to position the Button or Toolbar User Interfaces on the page.
+      If omitted, those User Interfaces will appear at the top of the document body.
+      (It is ignored by other User Interfaces.)
+    -->
+    <div id='KeymanWebControl'></div>                                                                                   
+
+    <h3>Type in your language in this text area:</h3>
```
Comment (darcywong00) [inline web/testing/toolbar/index.html]: Maybe  add a brief description this page is testing issue #1275

**Comment by mcdurdin** [comment]
Query on failed build?

**Comment by jahorton** [comment]
It was failing on the `git clean` problem.

**Pull request merged**